lib/ruby_lsp/listeners/definition.rb

```diff
@@ -16,18 +16,19 @@ class Definition
             Interface::LocationLink,
           )],
           global_state: GlobalState,
-          uri: URI::Generic,
+          document: Document,
           node_context: NodeContext,
           dispatcher: Prism::Dispatcher,
           typechecker_enabled: T::Boolean,
         ).void
       end
-      def initialize(response_builder, global_state, uri, node_context, dispatcher, typechecker_enabled) # rubocop:disable Metrics/ParameterLists
+      def initialize(response_builder, global_state, document, node_context, dispatcher, typechecker_enabled) # rubocop:disable Metrics/ParameterLists
         @response_builder = response_builder
         @global_state = global_state
         @index = T.let(global_state.index, RubyIndexer::Index)
         @type_inferrer = T.let(global_state.type_inferrer, TypeInferrer)
-        @uri = uri
+        @document = document
```

Comment:
Yes, for the server I believe that's a better idea because Sorbet will narrow types and help us get more out of the type checker.

But in this specific scenario, I really want to avoid passing document down. Maybe we can pass the document's class rather than the instance? Then we can differentiate the behaviour based on that.

Reply:
I ended up making `Document#language_id` returning an enum, which will get us the benefit of Sorbet in all cases and also save us from passing document here.